crates/noirc_evaluator/src/ssa/context.rs

```diff
@@ -30,7 +33,14 @@ pub struct SsaContext {
     value_names: HashMap<NodeId, u32>,
     pub sealed_blocks: HashSet<BlockId>,
     pub mem: Memory,
+
     pub functions: HashMap<FuncId, function::SSAFunction>,
+    pub function_ids: HashMap<FuncId, NodeId>,
```

Comment:
I would rather add the nodeid into the ssa_function structure and get rid of this map.